src/index.ts

```diff
@@ -125,10 +125,15 @@ async function getLocalEdgeConfig(
  * Edge Config Client
  */
 export interface EdgeConfigClient {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  get: <T = any>(key: string) => Promise<T | undefined>;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  getAll: <T = any>(keys?: (keyof T)[]) => Promise<T | undefined>;
+  get: <T extends EdgeConfigValue = EdgeConfigValue>(
+    key: string,
+  ) => Promise<T | undefined>;
+  getAll: <
+    T extends EdgeConfigValue = EdgeConfigValue,
+    K extends string = string,
+  >(
+    keys?: K[],
+  ) => Promise<Partial<Record<K, T>> | undefined>;
```

Comment:
one change I didn't make, but I think would be more accurate:
```suggestion
  ) => Promise<Partial<Record<K, T>>>;
```

It doesn't look like it's even possible to make an edge config `undefined` (I tried in the Vercel dashboard, not sure about through the API). So the response would always be at least `{}`.

But I wasn't 100% sure here, so I didn't make this change